codexctl/__init__.py

```
@@ -95,24 +100,50 @@ def call_func(self, function: str, args: dict) -> None:
+                        os.makedirs(output_dir, exist_ok=True)
+
+                    extract_cpio_files(args["file"], output_dir=output_dir)
+                    logger.info(f"Extracted SWU contents to {output_dir}")
+                else:
+                    logger.info("Detected old format (<3.11 .signed file)")
+                    try:
+                        from .analysis import get_update_image
+                    except ImportError:
+                        raise ImportError(
+                            "remarkable_update_image is required for extracting old format files. Please install it!"
+                        )
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)